ipaclient/install/client.py

```diff
@@ -3332,6 +3332,22 @@ def _install(options, tdict):
 
     if options.conf_ssh:
         configure_ssh_config(fstore, options)
+        # Make sure ssh service is enabled in the SSSD.conf file
+        try:
+            sssdconfig = SSSDConfig.SSSDConfig()
+            sssdconfig.import_config()
+            service = sssd_enable_service(sssdconfig, 'ssh')
+            if service is not None:
+                sssdconfig.save_service(service)
+                sssdconfig.write(paths.SSSD_CONF)
+            else:
+                logger.info(
+                'Failed to enable ssh service in the SSSD configuration file'
+                'sss_ssh_knownhostsproxy and sss_ssh_knownhosts tools require'
+                'sssd_ssh responder running to work properly'
+                )
+        except Exception as e:
+            logger.debug('Unable to update SSSD configuration: %s', str(e))
```

Comment:
I think it would be simpler to just do an enable_service() earlier in configure_sssd_conf() where the pam and nss services are enabled. It would be a two-liner:
```
if options.conf_ssh:
    sssd_enable_service(sssdconfig, 'ssh')
```
We don't, for good or bad, we only check the result of sssd_enable_service() with the ifp service and that is probably because support wasn't in all versions. I think the ssh service has been around forever. I don't think you'd need to do that here.
The writing and saving of the updated configuration file would happen naturally when the rest of the configuration is completed.

Reply:
I do agree @rcritten, amended .